assignment2/q1.py

```diff
@@ -0,0 +1,36 @@
+# Print AncestorsGiven a Binary Tree and a key, 
+# write a function that prints all the ancestors of the key in the givenbinary tree.
+class Node:
+ 
+    
+    def __init__(self, data):
+        self.data = data 
+        self.left = None
+        self.right = None
+
+    def printAncestors(root, target):
+
```

Comment:
@luxiya01, if you ask about the self.function() thing, it is a way of calling methods from within a class.

Aside the errors, the brackets in if are redundant. I recommend a liner or IDE like PyCharm to show those things, I'd add semicolons or something constanly  without them :D